assets/app/scripts/services/hpa.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("HPAService", function(LimitRangesService) {
+    var getCPURequestToLimitPercent = function(project) {
+      return LimitRangesService.getRequestToLimitPercent('cpu', project);
+    };
+
+    // Converts a percentage of request to a percentage of the limit based on
+    // the request-to-limit ratio.
+    // If CPU request percent is 120% and CPU request-to-limit percent is 50%, returns 60%
+    var convertRequestPercentToLimit = function(requestPercent, project) {
+      if (!requestPercent) {
+        return requestPercent;
+      }
+
+      var cpuRequestToLimitPercent = getCPURequestToLimitPercent(project);
+      var limitPercent = (cpuRequestToLimitPercent / 100) * requestPercent;
+      return Math.round(limitPercent);
+    };
+
+    // Converts a percentage of limit to a percentage of the request based on
+    // the request-to-limit ratio.
+    // If CPU limit percent is 60% and CPU request-to-limit percent is 50%, returns 120%
+    var convertLimitPercentToRequest = function(limitPercent, project) {
+      if (!limitPercent) {
+        return limitPercent;
+      }
+
+      var cpuRequestToLimitPercent = getCPURequestToLimitPercent(project);
+      var requestPercent = limitPercent / (cpuRequestToLimitPercent / 100);
+      return Math.round(requestPercent);
+    };
+
+    // Checks if any container has a value set for the compute resource request or limit.
+    //
+    // computeResource  - 'cpu' or 'memory'
+    // requestsOrLimits - 'requests' or 'limits'
+    // containers       - array of containters from a deployment config or replication controller
+    var hasRequestOrLimit = function(computeResource, requestsOrLimits, containers) {
+      return _.some(containers, function(container) {
+        return _.get(container, ['resources', requestsOrLimits, computeResource]);
+      });
+    };
+
+    var hasRequestSet = function(computeResource, containers) {
+      return hasRequestOrLimit(computeResource, 'requests', containers);
+    };
+
+    var hasLimitSet = function(computeResource, containers) {
+      return hasRequestOrLimit(computeResource, 'limits', containers);
+    };
+
+    // Checks if there's a default for the compute resource request or limit in any LimitRange.
+    //
+    // computeResource  - 'cpu' or 'memory'
+    // defaultType      - 'defaultRequest' or 'defaultLimit'
+    // limitsRanges     - collection of LimitRange objects (hash or array)
+    var hasDefault = function(computeResource, defaultType, limitRanges) {
+      // Check each limit range.
+      return _.some(limitRanges, function(limitRange) {
```

Comment:
LimitRangesService.getEffectiveLimitRange ?  there has been some discussion about limit ranges possibly having scopes like we did with quotas, in which case i think getEffectiveLimitRange would be updated to take into account scopes.